src/Pomelo.EntityFrameworkCore.MySql/Migrations/MySqlMigrationsSqlGenerator.cs

```diff
@@ -94,50 +94,70 @@ protected override void Generate(AlterColumnOperation operation, IModel model, M
                 .Append(operation.IsNullable ? " NULL" : " NOT NULL")
                 .AppendLine(SqlGenerationHelper.StatementTerminator);
 
-            alterBase = $"ALTER TABLE {identifier} ALTER COLUMN {SqlGenerationHelper.DelimitIdentifier(operation.Name)}";
-
-            builder.Append(alterBase);
-
-            if (operation.DefaultValue != null)
-            {
-                builder.Append(" SET DEFAULT ")
-                    .Append(SqlGenerationHelper.GenerateLiteral((dynamic)operation.DefaultValue))
-                    .AppendLine(SqlGenerationHelper.BatchTerminator);
-            }
-            else if (!string.IsNullOrWhiteSpace(operation.DefaultValueSql))
+            switch (type)
             {
-                builder.Append(" SET DEFAULT ")
-                    .Append(operation.DefaultValueSql)
-                    .AppendLine(SqlGenerationHelper.BatchTerminator);
-            }
-            else if (isSerial)
-            {
-                builder.Append(" SET DEFAULT ");
+                case "tinyblob":
+                case "blob":
+                case "mediumblob":
+                case "longblob":
+                case "tinytext":
+                case "text":
+                case "mediumtext":
+                case "longtext":
+                case "geometry":
```

Comment:
For geometry, include all of the [spatial data types](https://dev.mysql.com/doc/refman/5.7/en/spatial-datatypes.html)

- geometry
- point
- linestring
- polygon
- multipoint
- multilinestring
- multipolygon
- geometrycollection